shelf/endpoint_decorators.py

```diff
@@ -203,40 +208,47 @@ def wrapper(container, *args, **kwargs):
 
         return wrapper
 
-    def decode_request(self, container):
+    def decode_request(self, container, default_request_body=None):
         """
             Decodes data from flask request.
             Only accepts array or object as valid JSON.
 
             Args:
                 container(shelf.container.Container)
+                default_request_body(dict)
 
             Returns:
                 object | None: decoded JSON from request. None if invalid.
         """
-        data = container.request.get_json(silent=True, force=True)
-
-        if not isinstance(data, (list, dict)):
-            container.context.add_error(ErrorCode.INVALID_REQUEST_DATA_FORMAT)
-            data = None
+        if container.request.data:
+            data = container.request.get_json(silent=True, force=True)
+
+            if not isinstance(data, (list, dict)):
+                container.context.add_error(ErrorCode.INVALID_REQUEST_DATA_FORMAT)
+                data = None
+        else:
+            # Since Python passes objects by reference, we need to do a
+            # deep copy so it won't assign "data" to the
+            # reference of "default_request_body".
+            data = deepcopy(default_request_body)
 
         return data
 
-    def validate_request(self, schema_path):
+    def validate_request(self, schema_path, default_request_body=None):
         """
-            Decodes and validates request data against schema.
-            Is meant to be used after decode_request.
+            Validates request data against schema.
```

Comment:
This description is insufficient. It decodes data as JSON, injects data, and explain the whole default body thing as well.